skimage/_shared/utils.py

```diff
@@ -237,13 +237,21 @@ def convert_to_float(image, preserve_range):
         using img_as_float. Also see
         https://scikit-image.org/docs/dev/user_guide/data_types.html
 
+    Notes:
+    ------
+    * Input images with `float32` data type are not upcasted.
+
     Returns
     -------
     image : ndarray
         Transformed version of the input.
+
     """
     if preserve_range:
-        image = image.astype(np.double)
+        # Convert image to double only if it is not single or double
+        # precision float
+        if image.dtype.char not in 'df':
+            image = image.astype(np.double)
```

Comment:
```suggestion
            image = image.astype(float)
```

(This should be the default floating point type for the platform, ie whatever np.array([1.0]).dtype returns on that platform. I think using `float` ensures that.)